pkg/controller/handlers/accesscontrolrule/helper.go

```diff
@@ -0,0 +1,109 @@
+package accesscontrolrule
+
+import (
+	"fmt"
+
+	v1 "github.com/obot-platform/obot/pkg/storage/apis/obot.obot.ai/v1"
+	"github.com/obot-platform/obot/pkg/system"
+	gocache "k8s.io/client-go/tools/cache"
+)
+
+type Helper struct {
+	acrInformer gocache.SharedIndexInformer
+}
+
+func NewAccessControlRuleHelper(acrInformer gocache.SharedIndexInformer) *Helper {
+	return &Helper{
+		acrInformer: acrInformer,
+	}
+}
+
+// GetAccessControlRulesForUser returns all AccessControlRules that contain the specified user ID
+func (h *Helper) GetAccessControlRulesForUser(namespace, userID string) ([]v1.AccessControlRule, error) {
+	acrs, err := h.acrInformer.GetIndexer().ByIndex("user-ids", userID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get access control rules for user: %w", err)
+	}
+
+	result := make([]v1.AccessControlRule, 0, len(acrs))
+	for _, acr := range acrs {
+		res := acr.(*v1.AccessControlRule)
+		if res != nil && res.Namespace == namespace {
```

Comment:
Same comment here and the places below.

It's not possible for `res == nil` here. If `acr == nil`, then `arc.(*v1.AccessControlRule)` will panic. If this is something we want to protect against, then we can do:
```suggestion
		res, ok := acr.(*v1.AccessControlRule)
		if ok && res.Namespace == namespace {
```

Reply:
Fixed